### ISSUE TITLE
[Techdocs] Diff docs are not readable in dark mode

### ISSUE THREAD

**Issue by dtuite** [backstage/backstage#9395]
## Expected Behavior

I should be able to read diff docs 😄 

Something like GitHub would be perfect.

![Screenshot 2022-02-07 at 16 29 36](https://user-images.githubusercontent.com/562403/152830026-2517ad62-a49f-4706-8760-5211daa013c2.png)

## Current Behavior

It looks like this at the moment.

![Screenshot 2022-02-07 at 16 42 43](https://user-images.githubusercontent.com/562403/152832346-d3131fa8-813c-4780-aa75-3389d9fda38f.png)

## Possible Solution

Make the text dark when it's on a green or red background. Also, make the green and red background color somewhat opaque.

## Steps to Reproduce

Create a doc with this content (wrapped in three backticks followed by `diff`).

```diff
diff --git a/src/components/CallToAction/EmailCaptureForm.js b/src/components/CallToAction/EmailCaptureForm.js
index eb154ca..9450b93 100644
--- a/src/components/CallToAction/EmailCaptureForm.js
+++ b/src/components/CallToAction/EmailCaptureForm.js
@@ -56,7 +56,7 @@ const EmailCaptureForm = ({
           <Button
             text={buttonText}
             disabled={disabled}
-            icon={<PaperAirplaneIcon className="h-4 w-4 mr-1" />}
+            prefixIcon={<PaperAirplaneIcon />}
             id={buttonId}
             fullWidth
             color="primary"
```

## Context

It makes it difficult to show code changes in documentation.

**Comment by Irma12** [comment]
I would be interested in trying to make a progress here, so please assign it to me 😃

**Comment by Rugvip** [comment]
Fixed by #9476 🎉, thank you!
